control-plane/agents/src/bin/core/volume/service.rs

```diff
@@ -608,7 +608,10 @@ impl Service {
             return volume.resize(&self.registry, request).await;
         };
 
-        let required = request.requested_size - volume.as_ref().size;
+        let required = request
```

Comment:
This could do with a comment explaining how this handles the shrink case. I suppose required becomes zero?

Reply:
> This could do with a comment explaining how this handles the shrink case. I suppose required becomes zero?

Sure, added a comment. Yes it becomes zero. We could have added special handling to capacity handling for this case but this looks good with minimal change.